.github/workflows-helpers/run-e2e-test.sh

```diff
@@ -1,19 +1,28 @@
 # TODO_TECHDEBT(@okdas): also check readiness of appgate and relayminer to avoid false negatives due to race-conditions
 
-# Check if the pod with the matching image SHA and purpose is ready
-echo "Checking for ready validator pod with image SHA ${IMAGE_TAG}..."
+# Check if the pod with the matching image SHA and purpose is ready or needs recreation
+echo "Checking for ready validator pod with image SHA ${IMAGE_TAG} or pods needing recreation..."
 while :; do
     # Get all pods with the matching purpose
     PODS_JSON=$(kubectl get pods -n ${NAMESPACE} -l pokt.network/purpose=validator -o json)
 
     # Check if any pods are running and have the correct image SHA
     READY_POD=$(echo $PODS_JSON | jq -r ".items[] | select(.status.phase == \"Running\") | select(.spec.containers[].image | contains(\"${IMAGE_TAG}\")) | .metadata.name")
+    
+    # Check for non-running pods with incorrect image SHA to delete
+    INCORRECT_POD=$(echo $PODS_JSON | jq -r ".items[] | select(.status.phase != \"Running\") | select(.spec.containers[].image | contains(\"${IMAGE_TAG}\") | not) | .metadata.name")
```

Comment:
NIT: What do you think of adding an intermediary helper variable called `NON_RUNNING_PODS` so it's easier to read/nuderstand/main?